src/org/omegat/gui/glossary/GlossarySearcher.java

```diff
@@ -150,8 +150,7 @@ private List<Token[]> getMatchingTokens(Token[] fullTextTokens, String fullText,
 
     private static boolean rawMatch(Token[] tokens, String srcTxt, String term) {
         for (Token token : tokens) {
-            if (term.length() > token.getLength()
-                    && token.getTextFromString(srcTxt).equals(term.substring(token.getLength()))) {
+            if (term.length() > token.getLength() && term.contains(token.getTextFromString(srcTxt))) {
```

Comment:
We don't need to check length

```suggestion
            if (term.contains(token.getTextFromString(srcTxt))) {
```